### PR TITLE
BAEL-5348: Invert a Map in Java + tests

### PR DESCRIPTION
**Pull request by palani-a**
Article draft: https://drafts.baeldung.com/wp-admin/post.php?post=127508&action=edit 

I have fixed my repo and reverted the PR and commits related to my initial evaluation article (Hexagonal Architecture). This PR contains only changes to 2 files: InvertHashMapExample.java & InvertHashMapUnitTest.java.

Thanks
Palani

### CODE REVIEW

File: java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        sourceMap = new HashMap<>();
+        sourceMap.put("Sunday", 0);
+        sourceMap.put("Monday", 1);
+    }
+
+    @Test
+    void test1_invertMapUsingForLoop() {
+        Map<Integer, String> inversedMap = InvertHashMapExample.invertMapUsingForLoop(sourceMap);
+        assertNotNull(inversedMap);
+        assertEquals(inversedMap.size(), 2);
```
Comment (KevinGilmore) [inline java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java:31]: A better assertion here (and in the other tests) would be:
`assertEquals(sourceMap.size(), inversedMap.size());`
That way it's valid even if you add elements to your sourceMap in the test setup or before doing the inversion.

File: java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        assertEquals("Monday", inversedMap.get(1));
+    }
+
+    @Test
+    void test4_invertMapUsingGroupingBy() {
+        sourceMap.put("monday", 1);
+        Map<Integer, List<String>> inversedMap = InvertHashMapExample.invertMapUsingGroupingBy(sourceMap);
+        assertNotNull(inversedMap);
+        assertEquals(inversedMap.size(), 2);
+        assertTrue(inversedMap.get(1) instanceof List);
```
Comment (KevinGilmore) [inline java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java:57]: You have declared the values to be `List<String>` so this test isn't telling us anything about the result. A better assertion would be:
`assertEquals(2, inversedMap.get(1));`
That would, in a way, show that the duplicates were handled. An even better test would be to assert on the contents of the list (although you won't necessarily know the order of elements). The AssertJ library has some good assertions methods for dealing with these kinds of tests.

File: java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    void setup() {
+        sourceMap = new HashMap<>();
+        sourceMap.put("Sunday", 0);
+        sourceMap.put("Monday", 1);
+    }
+
+    @Test
+    void test1_invertMapUsingForLoop() {
+        Map<Integer, String> inversedMap = InvertHashMapExample.invertMapUsingForLoop(sourceMap);
+        assertNotNull(inversedMap);
```
Comment (KevinGilmore) [inline java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java:30]: add a blank line before beginning the assertions (applies to all the tests)

**Review by KevinGilmore — CHANGES_REQUESTED** [review]
See the test class for details.

File: java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class InvertHashMapUnitTest {
+
+    Map<String, Integer> sourceMap;
+
+    @BeforeAll
+    void setup() {
+        sourceMap = new HashMap<>();
+        sourceMap.put("Sunday", 0);
+        sourceMap.put("Monday", 1);
```
[the remark this answers is not in the document]
Comment (palani-a) [reply java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java:25]: Populated all 7 days.

File: java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java (hunk 3 above)
Comment (palani-a) [reply java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java:37]: Added blank line before beginning the assertions.

File: java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class InvertHashMapUnitTest {
+
+    Map<String, Integer> sourceMap;
+
+    @BeforeAll
```
[the remark this answers is not in the document]
Comment (palani-a) [reply java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java]: Used BeforeEach annotation as suggested.

File: java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java (hunk 1 above)
Comment (palani-a) [reply java-collections-maps-3/src/test/java/com/baeldung/map/invert/InvertHashMapUnitTest.java]: Incorporated the review comments.

**Review by KevinGilmore — APPROVED** [review]

**Pull request merged**